### PR TITLE
Handle unauthorized exceptions in a custom handler

### LINKED ISSUE #1254: Can't configure authentication on token API
**Version**
2.14.3

**Describe the bug**
Automation Hub relies on being able to configure the default DRF authentication classes in order to be able to provide podman and docker authentication when keycloak is configured. 

**Expected behavior**
The token API on the container registry is hard coded to use basic auth and can't be reconfigured

### PR DESCRIPTION
**Pull request by lubosmj**
closes #1254

### CODE REVIEW

File: pulp_container/app/settings.py [hunk 1]
```diff
@@ -1,4 +1,11 @@
+from django.conf import settings
+
 DRF_ACCESS_POLICY = {
     "dynaconf_merge_unique": True,
     "reusable_conditions": ["pulp_container.app.global_access_conditions"],
 }
+
+REST_FRAMEWORK = settings.REST_FRAMEWORK
+REST_FRAMEWORK.update(
+    {"EXCEPTION_HANDLER": "pulp_container.app.exceptions.unauthorized_exception_handler"}
```
Comment (newswangerd) [inline pulp_container/app/settings.py:10]: Won't this set the exception handler for all plugins?

Comment (lubosmj) [reply pulp_container/app/settings.py:10]: Hmm, that is a good question. I think it probably will. 😭

Comment (lubosmj) [reply pulp_container/app/settings.py:10]: I can try cloning this to make it available only to pulp_container.

File: pulp_container/app/settings.py [hunk 2]
```diff
@@ -1,4 +1,12 @@
...
+from django.conf import settings
+
 DRF_ACCESS_POLICY = {
     "dynaconf_merge_unique": True,
     "reusable_conditions": ["pulp_container.app.global_access_conditions"],
 }
+
+REST_FRAMEWORK = deepcopy(settings.REST_FRAMEWORK)
+REST_FRAMEWORK.update(
+    {"EXCEPTION_HANDLER": "pulp_container.app.exceptions.unauthorized_exception_handler"}
```
Comment (rochacbruno) [inline pulp_container/app/settings.py:11]:
```suggestion
DRF_ACCESS_POLICY__reusable_conditions = "@merge_unique pulp_container.app.global_access_conditions"

REST_FRAMEWORK__EXCEPTION_HANDLER = "pulp_container.app.exceptions.unauthorized_exception_handler"
```

File: pulp_container/app/settings.py [hunk 3]
```diff
@@ -1,4 +1,12 @@
+from copy import deepcopy
+from django.conf import settings
```
Comment (rochacbruno) [inline pulp_container/app/settings.py:2]: I don' t recommend importing `django.conf.settings` here, you can use dynaconf inline merging with `@ ` and  `__` (as in my other comment)

Or you can add a `pulp_container/app/dynaconf_hooks.py` and then do any conditionals or updates there.

`dynaconf_hooks.py`
```py
def post(settings):
    data = {"dynaconf_merge": True}
    data["REST_FRAMEWORK"]["EXCEPTION_HANDLER"] = "..."
    return data
```

The reason is that as settings is formed by the merging of multiple sources the final state is not the one you get grom django.conf at this point here.

**Pull request closed without merging**